### PR TITLE
When a reorg happens the wallet removes the transactions that are par…

### LINKED ISSUE #399: [Wallet-POS] When mining, wallet is able to lose track of coins
Started mining on testnet with 500k coins, after some time I only have 340k coins

### PR DESCRIPTION
**Pull request by dangershony**
…t of the reorg, however the spent trx where not marked as spendable again.

Fixes #399

### CODE REVIEW

**Comment by dangershony** [comment]
@Aprogiena

**Comment by Aprogiena** [comment]
Yes, exactly.

File: Stratis.Bitcoin.Features.Wallet/WalletManager.cs [hunk 1]
```diff
@@ -591,9 +591,15 @@ public void RemoveBlocks(ChainedBlock fork)
                 var allAddresses = this.keysLookup.Values;
                 foreach (var address in allAddresses)
                 {
+					// remove all the UTXO that have been reorged.
```
Comment (Aprogiena) [inline Stratis.Bitcoin.Features.Wallet/WalletManager.cs]: Tabs, you are adding tabs to the code.
Also inline comment style pls.
Here and below as well (both tabs and comment style)

File: Stratis.Bitcoin.Features.Wallet/WalletManager.cs [hunk 2]
```diff
@@ -712,18 +718,19 @@ public void ProcessTransaction(Transaction transaction, int? blockHeight = null,
...
-        /// <param name="transactionHash">The transaction hash.</param>
-        /// <param name="time">The time.</param>
-        /// <param name="index">The index.</param>
-        /// <param name="amount">The amount.</param>
-        /// <param name="script">The script.</param>
-        /// <param name="blockHeight">Height of the block.</param>
-        /// <param name="block">The block containing the transaction to add.</param>
-        /// <param name="transactionHex">The hexadecimal representation of the transaction.</param>
-        private void AddTransactionToWallet(string transactionHex, uint256 transactionHash, uint time, int index, Money amount, Script script,
+		/// <summary>
```
Comment (Aprogiena) [inline Stratis.Bitcoin.Features.Wallet/WalletManager.cs]: Tabs again, don't add tabs.

**Comment by dangershony** [comment]
@bokobza can you please review this as well?

**Comment by Aprogiena** [comment]
I'm going to merge this despite the fact that I was unable to reproduce the reorg. I will continue to try to reproduce it. But the code itself looks good to me.

Not really happy about it as I feel we progress little faster than we should, which causes problems like tests going red again, or that we introduce bugs to our code, or not having perfect styles ... However, I'm trying for more than 24 hours to reproduce the situation and I have simply no luck here. So I'll just keep it in mind and watch it closely, but move forward.

**Pull request merged**